### PR TITLE
PLAP-503 Generate REST API documentation using OpenAPI 3 and ReDoc.

### PR DESCRIPTION
**Pull request by cmanios**
- This will support the "API first" approach.

### CODE REVIEW

**Review by dimitriosprapas — APPROVED** [review]

**Review by nstergiou-prodyna — APPROVED** [review]

File: scripts/generate-redoc-html.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# This script converts OpenAPI 3.x yaml files to ReDoc html
+# using https://github.com/Redocly/redoc redoc-cli and then
+# generates an index.html web page using Gomplate template
+# renderer.
+
+set -eu
+
+ROOT_DIR="/opt/openapi-docs"
+OUT_DIR="/opt/oas3-dist"
+SVC_NAME_REGEX="^(.*)[.](ya?ml)$" && \
```
Comment (mkruse83) [inline scripts/generate-redoc-html.sh:12]: unused?

File: scripts/generate-redoc-html.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+cd "$ROOT_DIR"
+OLDIFS="$IFS"
```
Comment (mkruse83) [inline scripts/generate-redoc-html.sh:15]: unused?

File: scripts/generate-redoc-html.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+mkdir -p "$OUT_DIR"
+
+echo "services:" > "${OUT_DIR}/services.yml"
+
+find * -prune -type f | while IFS= read -r SVC_FILE; do
+
+    echo "Processing ${SVC_FILE} ..."
+
+    # Extract filename and suffix
```
Comment (mkruse83) [inline scripts/generate-redoc-html.sh:25]: please remove if not needed

File: scripts/generate-redoc-html.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    # Extract filename and suffix 
+    # (this solution works only in /bin/bash)
+    #[[ ${SVC_FILE} =~ ${SVC_NAME_REGEX} ]] && \
+    #SVC_NAME="${BASH_REMATCH[1]}" && \
+    #SVC_FILE_SUFFIX="${BASH_REMATCH[2]}"
+
+    # Extract filename and suffix
+    # (works in /bin/sh and also in /bin/bash)
+    SVC_NAME=$(basename -- "$SVC_FILE")
+    SVC_FILE_EXTENSION="${SVC_FILE##*.}"
```
Comment (mkruse83) [inline scripts/generate-redoc-html.sh:34]: unused?

File: scripts/generate-redoc-html.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+SVC_NAME_REGEX="^(.*)[.](ya?ml)$" && \
+
+cd "$ROOT_DIR"
+OLDIFS="$IFS"
+
+mkdir -p "$OUT_DIR"
+
+echo "services:" > "${OUT_DIR}/services.yml"
+
+find * -prune -type f | while IFS= read -r SVC_FILE; do
```
Comment (mkruse83) [inline scripts/generate-redoc-html.sh:21]:
```suggestion
find ./* -prune -type f | while IFS= read -r SVC_FILE; do
```

File: scripts/validate-oas.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# This script validates OpenAPI 3.x yaml files using
+# https://github.com/APIDevTools/swagger-cli.
+
+set -eu
+
+ROOT_DIR="/opt/openapi-docs"
+OUT_DIR="/opt/oas3-dist"
```
Comment (mkruse83) [inline scripts/validate-oas.sh:9]: unused?

File: scripts/validate-oas.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+echo "Listing ${ROOT_DIR}"
+ls -l "$ROOT_DIR" 
+cd "$ROOT_DIR"
+OLDIFS="$IFS"
```
Comment (mkruse83) [inline scripts/validate-oas.sh:14]: unused?

File: scripts/validate-oas.sh [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+find * -prune -type f | while IFS= read -r SVC_FILE; do
```
Comment (mkruse83) [inline scripts/validate-oas.sh:16]:
```suggestion
find ./* -prune -type f | while IFS= read -r SVC_FILE; do
```

**Review by mkruse83 — APPROVED** [review]
Only some minor cleanups recommended. Thanks for the effort :)

File: scripts/generate-redoc-html.sh (hunk 1 above)
Comment (cmanios) [reply scripts/generate-redoc-html.sh]: Hi @mkruse83 !

The change has been added in d5082b1b395f473bc5b5c41f3e4686d6b52be333.

Thanks!

File: scripts/generate-redoc-html.sh (hunk 2 above)
Comment (cmanios) [reply scripts/generate-redoc-html.sh]: Hi @mkruse83 !

The change has been added in d5082b1b395f473bc5b5c41f3e4686d6b52be333.

Thanks!

File: scripts/generate-redoc-html.sh (hunk 3 above)
Comment (cmanios) [reply scripts/generate-redoc-html.sh]: Hi @mkruse83 !

The change has been added in d5082b1b395f473bc5b5c41f3e4686d6b52be333.

Thanks!

File: scripts/generate-redoc-html.sh (hunk 4 above)
Comment (cmanios) [reply scripts/generate-redoc-html.sh]: Hi @mkruse83 !

The change has been added in d5082b1b395f473bc5b5c41f3e4686d6b52be333.

Thanks!

File: scripts/generate-redoc-html.sh (hunk 5 above)
Comment (cmanios) [reply scripts/generate-redoc-html.sh]: Hi @mkruse83 !

The change has been added in d5082b1b395f473bc5b5c41f3e4686d6b52be333.

Thanks!

File: scripts/validate-oas.sh (hunk 6 above)
Comment (cmanios) [reply scripts/validate-oas.sh]: Hi @mkruse83 !

The change has been added in d5082b1b395f473bc5b5c41f3e4686d6b52be333.

Thanks!

File: scripts/validate-oas.sh (hunk 7 above)
Comment (cmanios) [reply scripts/validate-oas.sh]: Hi @mkruse83 !

The change has been added in d5082b1b395f473bc5b5c41f3e4686d6b52be333.

Thanks!

File: scripts/validate-oas.sh (hunk 8 above)
Comment (cmanios) [reply scripts/validate-oas.sh]: Hi @mkruse83 !

The change has been added in d5082b1b395f473bc5b5c41f3e4686d6b52be333.

Thanks!

**Pull request merged**